kitsune/src/http/pagination.rs

```diff
@@ -0,0 +1,71 @@
+use std::{convert::Infallible, fmt::Display};
+
+use axum::{
+    response::{IntoResponseParts, ResponseParts},
+    Json,
+};
+use http::HeaderValue;
+
+pub type PaginatedJsonResponse<T> = (Option<Link<Vec<(&'static str, String)>>>, Json<Vec<T>>);
+
+pub struct Link<T>(pub T);
+
+impl<T, K, V> IntoResponseParts for Link<T>
+where
+    T: IntoIterator<Item = (K, V)>,
+    K: Display,
+    V: Display,
+{
+    type Error = Infallible;
+
+    fn into_response_parts(self, mut res: ResponseParts) -> Result<ResponseParts, Self::Error> {
+        let value = self
+            .0
+            .into_iter()
+            .map(|(key, value)| format!("<{value}>; rel=\"{key}\""))
+            .collect::<Vec<String>>()
+            .join(", ");
+
+        // TODO: log an error when HeaderValue conversion fails
+        if let Ok(header_value) = HeaderValue::from_str(&value) {
+            res.headers_mut().insert("Link", header_value);
+        }
+
+        Ok(res)
+    }
+}
+
+pub fn new_link_header<I, D: Display, F: Fn(&I) -> D>(
+    collection: &Vec<I>,
```

Comment:
Using a slice here is more idiomatic 

```suggestion
    collection: &Vec<I>,
    collection: &[I],
```

Reply:
You're absolutely right, I fixed that